datasets/openwebtext/openwebtext.py

```diff
@@ -0,0 +1,86 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""The Open WebText Corpus"""
+
+from __future__ import absolute_import, division, print_function
+
+import os
+from itertools import chain
+import re
+
+import datasets
+
+
+_CITATION = """\
+@misc{Gokaslan2019OpenWeb,  
+	title={OpenWebText Corpus},
+	author={Aaron Gokaslan*, Vanya Cohen*, Ellie Pavlick, Stefanie Tellex},
+	howpublished{\\url{http://Skylion007.github.io/OpenWebTextCorpus}}, 
+	year={2019}
+}
+"""
+
+_DESCRIPTION = """\
+An open-source replication of the WebText dataset from OpenAI.
+"""
+
+_URL = (
+    "https://zenodo.org/record/3834942/files/openwebtext.tar.xz"
+)
+
+
+class Openwebtext(datasets.GeneratorBasedBuilder):
+    """The Open WebText dataset."""
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            name="plain_text",
+            description="Plain text",
+            version=datasets.Version("1.0.0"),
+        )
+    ]
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features({"text": datasets.Value("string")}),
+            homepage="https://skylion007.github.io/OpenWebTextCorpus/",
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        dl_dir = dl_manager.download_and_extract(_URL)
+        owt_dir = os.path.join(dl_dir, 'openwebtext')
+        subset_xzs = [
+            os.path.join(owt_dir, file_name) for file_name in os.listdir(owt_dir) if file_name.endswith('xz') # filter out ...xz.lock
```

Comment:
I'm not sure that `os.listdir` always returns the files in the same order depending on the filesystem.
Maybe we should consider sorting the file_names ?

Reply:
I didn't be aware of whether the order of file names returned by `os.listdir`. I thought as long as it collect all files, it is ok. Is there anything we should care about the order of file names ?

Reply:
Ideally the dataset should be ordered the same way for everyone.
Since there's a loop on the files to generate examples we need the files to always have the same order

Reply:
Yeah you're right ! It should be sorted.